### PR TITLE
Include the inspect-component script.

### PR DESCRIPTION
**Pull request by frankban**

### CODE REVIEW

**Comment by jujugui** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://ci-cge.jujugui.org:8080//job/juju-gui/7358/

File: Makefile [hunk 1]
```diff
@@ -300,6 +300,11 @@ lint-js: $(NODE_MODULES)
 lint-css: $(NODE_MODULES)
 	$(NODE_MODULES)/.bin/sass-lint -q -v
 
+.PHONY: lint-components
+lint-components:
+	 @./scripts/inspect-components validate --path jujugui/static/gui/src/ --short || \
+	 	echo "^^^ Ping frankban if those are not actual errors.\n"
```
Comment (jaycee) [inline Makefile]: Nice. :smile:

Do we want to specify a channel for non canonical contributors? Or at this point are we just admitting that we're the only one's working on this? :stuck_out_tongue_closed_eyes:

File: scripts/inspect-components [hunk 2]
```diff
@@ -0,0 +1,488 @@
...
+    (?=  # Until the following happens.
+        propTypes\s*  # Properties are found.
+        :\s*  # Property definitions.
+        \{  # Brace opening.
+        ([\s\S]*?)  # Any character.
+        (?=  # Until the following happens.
+            \}  # A close brace is encountered.
+        )  # End of condition.
+    )  # End of condition.
+""", re.VERBOSE | re.MULTILINE)
```
Comment (jaycee) [inline scripts/inspect-components]: So this is all just witchcraft, right?

Less glibly, thanks for breaking this out with all the comments.

File: scripts/inspect-components [hunk 3]
```diff
@@ -0,0 +1,488 @@
...
+import operator
+import os
+import re
+import sys
+
+
+# Define global regular expressions.
+_definition_exp = re.compile(r"""
+    ([\w\._]+)\s*  # Component name.
+    =\s*  # Assignation.
```
Comment (jaycee) [inline scripts/inspect-components]: Assignment?

File: scripts/inspect-components [hunk 4]
```diff
@@ -0,0 +1,488 @@
...
+        \{  # Brace opening.
+        ([\s\S]*?)  # Any character.
+        (?=  # Until the following happens.
+            \}  # A close brace is encountered.
+        )  # End of condition.
+    )  # End of condition.
+""", re.VERBOSE | re.MULTILINE)
+_fields_exp = re.compile(r"""
+    ([\w_]+)\s*  # Field name.
+    =\s*  # Assignation.
```
Comment (jaycee) [inline scripts/inspect-components]: Again, I think you want "Assignment" rather than "Assignation".

File: Makefile (hunk 1 above)
Comment (frankban) [inline Makefile]: You are right, I slightly changed this.

File: scripts/inspect-components (hunk 2 above)
Comment (frankban) [inline scripts/inspect-components]: Yeah the verbose mode really helps!

File: scripts/inspect-components (hunk 4 above)
Comment (frankban) [inline scripts/inspect-components]: Shame... shame!

Comment (frankban) [inline scripts/inspect-components]: And fixed.

File: scripts/inspect-components (hunk 3 above)
Comment (frankban) [inline scripts/inspect-components]: I guess so...

**Comment by jujugui** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://ci-cge.jujugui.org:8080//job/juju-gui/7359/

**Comment by frankban** [comment]
Thanks for the review JC!
:shipit:

**Comment by jujugui** [comment]
Status: merge request accepted. Url: http://ci-gce.jujugui.org:8080/job/juju-gui-merge

**Pull request merged**